python/pyarrow/types.pxi

```diff
@@ -75,6 +75,11 @@ cdef bytes _datatype_to_pep3118(CDataType* type):
             return char
 
 
+def _is_primitive(Type type):
```

Comment:
Can't this be a DataType method or property instead?

Reply:
Yes, this could also be a DataType but as the comment suggests, I have just added this definition here so I can call the C++ function from Python. The user-facing `is_primitive` accepts a DataType.